internal/cli/organizations.go

```diff
@@ -461,38 +472,175 @@ auth0 orgs members ls <id>`,
+						}
 					}
-					return output, members.HasNext(), nil
-				})
-
+				}
+				return nil
+			})
 			if err != nil {
-				return fmt.Errorf("Unable to list members of an organization with Id '%s': %w", inputs.ID, err)
+				return err
 			}
-			var typedList []management.OrganizationMember
-			for _, item := range list {
-				typedList = append(typedList, item.(management.OrganizationMember))
+			// convert management.OrganizationMemberRole to management.Role
+			var roles []*management.Role
+			for _, role := range roleMap {
+				roles = append(roles, &management.Role{ID: role.ID, Name: role.Name, Description: role.Description})
 			}
-			cli.renderer.MembersList(typedList)
+			sort.Slice(roles, func(i, j int) bool {
+				return strings.ToLower(roles[i].GetName()) < strings.ToLower(roles[j].GetName())
+			})
+			cli.renderer.RoleList(roles)
 			return nil
 		},
 	}
+	return cmd
+}
+
+func membersRolesOrganizationCmd(cli *cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "members",
+		Short: "Manage roles of organization members",
+		Long:  "Manage roles assigned to members of an organization.",
+	}
+
+	cmd.SetUsageTemplate(resourceUsageTemplate())
+	cmd.AddCommand(listMembersRolesOrganizationCmd(cli))
 
 	return cmd
 }
 
-func (c *cli) organizationPickerOptions() (pickerOptions, error) {
-	list, err := c.api.Organization.List()
+func listMembersRolesOrganizationCmd(cli *cli) *cobra.Command {
+	var inputs struct {
+		OrgID  string
+		RoleID string
+		Number int
+	}
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Args:  cobra.MaximumNArgs(1),
+		Short: "List organization members for a role",
+		Long:  "List organization members that have a given role assigned to them.",
+		Example: `auth0 orgs roles members list
+auth0 orgs roles members list <org id> --role-id role`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				err := organizationID.Pick(cmd, &inputs.OrgID, cli.organizationPickerOptions)
+				if err != nil {
+					return err
+				}
+				if inputs.RoleID == "" {
+					err = roleID.Pick(cmd, &inputs.RoleID, cli.rolePickerOptions)
+					if err != nil {
+						return err
+					}
+				}
+			} else {
+				inputs.OrgID = args[0]
+			}
+
+			members, err := cli.getOrgMembersWithSpinner(cmd.Context(), inputs.OrgID, inputs.Number)
+			if err != nil {
+				return err
+			}
+
+			var roleMembers []management.OrganizationMember
+			errSpinner := ansi.Spinner("Getting roles assigned to organization members", func() error {
+				for _, member := range members {
+					userID := member.GetUserID()
+					roleList, err := cli.api.Organization.MemberRoles(inputs.OrgID, userID)
+					if err != nil {
+						return err
+					}
+					for _, role := range roleList.Roles {
+						roleID := role.GetID()
+						if roleID == inputs.RoleID {
+							roleMembers = append(roleMembers, member)
+						}
+					}
+				}
+				return nil
+			})
+			if errSpinner != nil {
+				return errSpinner
+			}
+			sortMembers(roleMembers)
```

Comment:
This looks like a candidate as well to be moved within a func so it's easier to navigate the codebase. Wdyt?